addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/NeeoHandlerCallback.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.neeo.internal;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.State;
+
+/**
+ *
+ * This interface is used to provide a callback mechanism between a @link {@link ThingHandler} and the assoicated
+ * protocol.
+ * This is necessary since the status and state of a bridge/thing is private and the protocol handler cannot access it
+ * directly.
+ *
+ * @author Tim Roberts - initial contribution
+ *
+ */
+public interface NeeoHandlerCallback {
+    /**
+     * Callback to the bridge/thing to update the status of the bridge/thing.
+     *
+     * @param status a non-null {@link ThingStatus}
```

Comment:
Someone needs to make a decision on this one - I used it a few times here or there and one review comment somewhere said it should either be all or none.  Since this started prior to the attributes - that would be a huge job and I opted for none.  If you don't mind a mix, I'll add it where it's interesting...